content/guide/cli-basics.md

```diff
@@ -0,0 +1,131 @@
+
+Cleaning a NativeScript project is similar to running "Clean Build Folder" in XCode or other IDE environments. Sometimes during development you may trouble. To achieve that, run:
+
+```cli
+ns clean
+```
+Running `ns clean` is similar to running "Clean Build Folder" in XCode or other IDE environments. It deletes the `node_modules`, `hooks`, and `platforms` directories from your project.
+
+### Running an app on a device or simulator
+
+To run your app on all connected devices and emulators, run:
+
+```cli
+ns run
+```
+
+The command prepares, builds and deploys the app on the connected devices. By default, it listens for changes in your code, synchronizes those changes and refreshes all the devices.
+
+To launch the app on a connected Android device or Android emulator, run:
+
+```cli
+ns run android
+```
+
+::: warning Note
+If you get an error at this point you might not have completed the full [NativeScript CLI setup](/setup/).
+
+You must have at least one AVD (Android Virtual Device) configured on your development machine for this command to run your app up on an Android emulator.
+Or a connected Android device with debugging enabled.
+
+Check the `devDependencies` of your `package.json` file. `@nativescript/android` must be installed to avoid the "[Unable to apply changes on device: emulator-5554. Error is: Invalid Version: null.](https://github.com/NativeScript/nativescript-cli/issues/4451)" error.
+:::
+
+To launch the app on a connected iOS device or iOS simulator.
+```cli
+ns run ios
+```
+
+::: tip Note
+NativeScript uses Xcode to build and run iOS apps, and Xcode is only available on macOS; therefore, you can only run iOS apps on macOS. There are VM and/or cloud services that allow you to build on a Mac from a PC.
+:::
+
+
+You can customize the `ns run` command using any of the following options:
+
+- `--no-hmr` - Disables the webpack HMR option, so changes made during a session will restart the application.
+- `--emulator` - Specifies that you want to debug the app in an emulator.
+- `--timeout` - Sets the number of seconds that the NativeScript CLI will wait for the debugger to boot. If not set, the default timeout is 90 seconds.
+- `--clean` - If set forces rebuilding the native application.
+
+::: tip Note
+If you see this output in the terminal:
+
+```
+Webpack compilation complete. Watching for file changes.
+Watchpack Error (watcher): Error: EMFILE: too many open files 'FILE_PATH'
+Watchpack Error (watcher): Error: EMFILE: too many open files 'FILE_PATH'
+Watchpack Error (watcher): Error: EMFILE: too many open files 'FILE_PATH' <-- This repeats many times
+```
+
+This is related to node configuration options on your machine.
+
+**Solution**:
+
+Try adding this to your `~/.bash_profile` if you have one or `~/.zshenv` if using Zsh:
+
+```
+export NODE_OPTIONS="--max-old-space-size=6096"
+```
+
+Then open a new terminal window and run your app.
+:::
```

Comment:
```suggestion
```

This section has been added to the troubleshooting page: https://beta.docs.nativescript.org/troubleshooting#emfile-too-many-open-files-file-path